lib/ecto/query.ex

```diff
@@ -1928,17 +1967,41 @@ defmodule Ecto.Query do
 
   If `limit` is given twice, it overrides the previous value.
 
+  ## Options
+
+    * `:with_ties` - a boolean controlling whether records tied for the last
+       spot will be returned. Not every adapter will support this option and
+       some adapters may require a corresponding `order_by` clause. Please 
+       check your database's documentation for more details. For SQL databases,
+       it may be found under the `FETCH` command instead of `LIMIT`. Defaults to `false`
+
   ## Keywords example
 
       from(u in User, where: u.id == ^current_user, limit: 1)
 
+      from(u in User, where: u.id == ^current_user, limit: 1, with_ties: true)
+
   ## Expressions example
 
       User |> where([u], u.id == ^current_user) |> limit(1)
 
+      User |> where([u], u.id == ^current_user) |> limit([], 1, with_ties: true)
+
   """
-  defmacro limit(query, binding \\ [], expr) do
-    Builder.LimitOffset.build(:limit, query, binding, expr, __CALLER__)
+  defmacro limit(query, binding \\ [], expr, opts \\ [])
+  defmacro limit(query, binding, expr, opts) when is_list(binding) and is_list(opts) do
+    with_ties = opts[:with_ties] || false
```

Comment:
This is assuming `opts` is a keyword at compile-time. This means this would break:

```elixir
opts = [with_ties: true]
limit(query, [], 10, opts)
```

You need to handle the default value inside `Builder.Limit` inside a quote expression. :)